bndtools.core/src/bndtools/command/BndrunFilesParameterValues.java

```diff
@@ -21,44 +23,40 @@
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IPath;
 
-import aQute.bnd.build.Workspace;
-import bndtools.central.Central;
+import aQute.lib.memoize.Memoize;
 
 public class BndrunFilesParameterValues implements IParameterValues {
 
 	static final String BNDRUN_FILE = "bnd.command.bndrunFile";
 
-	private static Map<String, String> bndrunFilesMap;
-	private static IResourceChangeListener	listener	= new InvalidateMapListener();
+	private IResourceChangeListener					listener	= new InvalidateMapListener();
+	private volatile Supplier<Map<String, String>>	bndrunFilesMap;
+
+	public BndrunFilesParameterValues() {
+		bndrunFilesMap = reset();
+	}
+
+	Supplier<Map<String, String>> reset() {
+		return Memoize.referenceSupplier(() -> {
+			Map<String, String> map = Arrays.stream(ResourcesPlugin.getWorkspace()
+				.getRoot()
+				.getProjects())
+				.filter(IProject::isOpen)
+				.map(this::findBndrunFiles)
+				.flatMap(Collection::stream)
+				.map(IResource::getFullPath)
+				.collect(toMap(IPath::toPortableString, IPath::toPortableString));
+
+			ResourcesPlugin.getWorkspace()
```

Comment:
Perhaps you don't want to add the listener in the supplier since you will get an additional listener every time reset is called and the new memoized is initialized. The each listener will further call reset...

I think you just need to add the listener in the constructor.

Reply:
fixed